File: translation_output/deepseekcoder/codenet/java/python/p00378.py
<｜begin▁of▁sentence｜>You are an expert Python programmer.Your task is to write a Python function to solve a programming problem.

### Instruction:
Translate the following java code to python and enclose your solution inside ```python```:
```
import java.util.*;
class p00378{
    void solve(){
        int A, B, X, ans;
        Scanner sc = new Scanner(System.in);
        A = sc.nextInt();
        B = sc.nextInt();
        X = sc.nextInt();

        if ( X % 500 != 0 ) X += 500 - X%500;
        int a = 0;
        int b = 0;

        if ( A < B ) {
            a = X/1000 + ((X%1000>0)?1:0);
        } else if ( A > 2*B ){
            b = X/500;
        } else {
            a = X/1000; X %= 1000;
            b = X/500;
        }

        System.out.println(A*a + B*b);
    }

    public static void main(String[] args){ new p00378().solve(); }
}


```

### Response:
```python
class p00378:
    def solve(self):
        A, B, X = map(int, input().split())

        if X % 500 != 0: X += 500 - X % 500
        a = b = 0

        if A < B:
            a = X // 1000 + (X % 1000 > 0)
        elif A > 2 * B:
            b = X // 500
        else:
            a = X // 1000; X %= 1000
            b = X // 500

        print(A * a + B * b)


if __name__ == "__main__":
    p00378().solve()
```

This Python code does the same thing as the Java code. It reads three integers from the input, adjusts the third integer to be a multiple of 500, and then calculates and prints the maximum possible total.
<|EOT|>

File: translation_output/misleading_test/deepseekcoder/codenet/python/java/p00378.java
<｜begin▁of▁sentence｜>You are an expert Python programmer.Your task is to write a Python function to solve a programming problem.

### Instruction:
Translate the following python code to java and enclose your solution inside ```java```.
A sample test case is provided below:

Test input:
180 100 2400

Expected output:
491

```
A, B, X = map(int, input().split())
k = (X // 1000)
ans = k * min(A, 2*B)
r = X - 1000*k
if 0 < r <= 500:
    ans += min(A, B)
elif 500 < r:
    ans += min(A, 2*B)
print(ans)

```

### Response:
Here is the Java version of the provided Python code:

```java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int A = scanner.nextInt();
        int B = scanner.nextInt();
        int X = scanner.nextInt();

        int k = X / 1000;
        int ans = k * Math.min(A, 2*B);
        int r = X - 1000*k;
        if (0 < r && r <= 500) {
            ans += Math.min(A, B);
        } else if (r > 500) {
            ans += Math.min(A, 2*B);
        }
        System.out.println(ans);
    }
}
```

This Java code does the same thing as the Python code. It reads three integers from the input, calculates the value of `k`, and then calculates the value of `ans` based on the conditions in the Python code. Finally, it prints the value of `ans`.
<|EOT|>
